### PR TITLE
generator: Support the latest package dependencies

### PR DESCRIPTION
**Pull request by kevmoo**
Also wired through the language version of the target `web` package

### CODE REVIEW

File: web_generator/lib/src/dart_main.dart [hunk 1]
```diff
@@ -19,15 +20,24 @@ import 'util.dart';
 // probably involve parsing the TC39 spec.
 
 void main(List<String> args) async {
+  var languageVersionString = const String.fromEnvironment('languageVersion');
+  if (languageVersionString == '') {
```
Comment (srujzs) [inline web_generator/lib/src/dart_main.dart:24]: nit: `isEmpty`

File: web_generator/bin/update_bindings.dart [hunk 2]
```diff
@@ -128,6 +131,25 @@ $_startComment
   }
 }
 
+Future<String> _webPackageLanguageVersion(String pkgPath) async {
```
Comment (srujzs) [inline web_generator/bin/update_bindings.dart:134]: IIUC, you want to format the `web` package using the language version that it is tied to within its pubspec?

Did you end up noticing lots of changes wrt formatting when you ran the generator before this PR?

**Review by srujzs — APPROVED** [review]

File: web_generator/mono_pkg.yaml [hunk 3]
```diff
@@ -9,8 +9,12 @@ stages:
   - test: --test-randomize-ordering-seed=random -p chrome
   - test: --test-randomize-ordering-seed=random -p chrome -c dart2wasm
 - generate_and_analyze:
-  - command: dart bin/update_bindings.dart
-  - analyze: --fatal-infos .
+  - command:
+    - dart pub -C ../web get
+    - dart bin/update_bindings.dart
+    - dart analyze --fatal-infos ../web
```
Comment (kevmoo) [inline web_generator/mono_pkg.yaml:15]: before we were analyzing `web_generator`! We want to analyze `web` after `update_bindings` right?

@srujzs

File: web_generator/lib/src/dart_main.dart (hunk 1 above)
Comment (kevmoo) [reply web_generator/lib/src/dart_main.dart]: yep

File: web_generator/bin/update_bindings.dart (hunk 2 above)
Comment (kevmoo) [reply web_generator/bin/update_bindings.dart:134]: yep! mostly comma removals, but there were some spacing things that also showed up

Comment (srujzs) [reply web_generator/bin/update_bindings.dart:134]: ack, makes sense. thanks!

File: web_generator/mono_pkg.yaml (hunk 3 above)
Comment (srujzs) [reply web_generator/mono_pkg.yaml:15]: Each repo has a `mono_pkg.yaml` file. Here, we're analyzing the `web_generator`, but there are separate jobs to analyze `web` in the `mono_pkg.yaml` file under `web`.

Comment (kevmoo) [reply web_generator/mono_pkg.yaml:15]: I know that! 😛 But we want to validate `web` after we run the `update_bindings` verification, which makes sense to have under `web_generator`

Comment (srujzs) [reply web_generator/mono_pkg.yaml:15]: Oh! Yes, it makes no sense to analyze `web_generator` after the generation. I added this specifically to get info on whether new IDL versions break us, so this LGTM.

**Pull request merged**